fe/src/main/java/org/apache/doris/catalog/Catalog.java

```diff
@@ -2639,6 +2641,7 @@ public void dropDb(DropDbStmt stmt) throws DdlException {
             fullNameToDb.remove(db.getFullName());
             final Cluster cluster = nameToCluster.get(db.getClusterName());
             cluster.removeDb(dbName, db.getId());
+            globalTransactionMgr.removeDatabaseTransactionMgr(db.getId());
```

Comment:
Actually, the operation `DropDb` is just put database to CatalogRecycleBin, not actually drop it. And the database can be recovered by `Recover` operation.
So here you should not `removeDatabaseTransactionMgr()`, Instead, it should be called in `CatalogRecycleBin.eraseDatabase()`